web/modules/custom/collection/src/ContextProvider/CollectionRouteContext.php

```diff
@@ -48,6 +51,22 @@ public function getRuntimeContexts(array $unqualified_context_ids) {
         }
       }
     }
+    elseif ($route_parameters = $this->routeMatch->getParameters()) {
+      /** @var ContentEntityInterface $route_parameter */
+      foreach ($route_parameters as $route_parameter) {
```

Comment:
This is strange, doesn't this mean that you would loop over all parts that the route consists of? So if there are multiple parameters in the route (e.g. in a route like /entity/123/other_entity/456) then it will check each of them and consider the route to be part of a collection if any of the entities match.

We should only check the actual route object, not the parameters.